src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1798,6 +1798,40 @@
       <summary>Creates an ImplicitStackAllocArrayCreationExpressionSyntax node.</summary>
     </FactoryComment>
   </Node>
+  <Node Name="CollectionExpressionSyntax" Base="ExpressionSyntax">
+    <Kind Name="CollectionExpression"/>
+    <Field Name="OpenBracketToken" Type="SyntaxToken">
+      <Kind Name="OpenBracketToken"/>
+    </Field>
+    <Field Name="Elements" Type="SeparatedSyntaxList&lt;CollectionElementSyntax&gt;" AllowTrailingSeparator="true">
+      <PropertyComment>
+        <summary>SeparatedSyntaxList of CollectionElementSyntax representing the list of elements in the collection expression.</summary>
+      </PropertyComment>
+    </Field>
+    <Field Name="CloseBracketToken" Type="SyntaxToken">
+      <Kind Name="CloseBracketToken"/>
+    </Field>
+  </Node>
+  <AbstractNode Name="CollectionElementSyntax" Base="CSharpSyntaxNode" />
```

Comment:
i also considered CollectionExpressionElementSyntax, but it was enormously wordy.  I considered ElementSyntax, but i also felt that was too vague as to what it corresponded to.  CollectionElementSyntax was a middle ground of making it clear this is about collections (and thus CollectionExpression).